### DISCUSSION TITLE
Missing `openjdk` man pages

### DISCUSSION THREAD

**Discussion by ghost** [Homebrew/discussions#1843] [Linux] [answered]
The man pages for `llvm` and `openjdk` are not installed. 
After some digging, I found the ones for openjdk at `/home/linuxbrew/.linuxbrew/Cellar/openjdk/<version>/libexec/man/man1` though they are not installed in the system. 
I could not trace llvm's. 
How can I get them all installed? They are really important offline documentation

**Comment by carlocab** [comment]
I opened Homebrew/homebrew-core#81457 to symlink the manpages into a more public location. You can also currently find them at `/home/linuxbrew/.linuxbrew/opt/openjdk/libexec/man/man1` so that you don't need to keep track of a path that changes with version/revision bumps.

We don't build LLVM's manpages on purpose. They introduce additional dependencies and add complexity to the build. You can find the documentation for each LLVM release at https://releases.llvm.org.

**Comment by ghost** [comment]
Thanks

**Comment by ghost** [comment, marked as answer]
Hi there again...that does not work at all

**Reply by cho-m** [reply]
Which part does not work? The new merged fix, the libexec manpage directory, or something else?

---

For merged fix, to check, are you on Linux with repo from `linuxbrew-core`?
If so, the bottle may need to be updated in that repo: https://github.com/Homebrew/linuxbrew-core/commit/0f1a9b3c3cfee9d01d191eeb7ef9c4d3900ef39e#diff-df00e346e6e2293ca1363f7b202030ab4e24569f520a49635bc29692e7eb5283

On Linux-side, I am helping with merger of `linuxbrew-core -> homebrew-core` and can confirm the Linux bottle in (the currently experimental) `homebrew-core` works with default directory from `brew shellenv`:
```
# ls /home/linuxbrew/.linuxbrew/share/man/man1/*java*
/home/linuxbrew/.linuxbrew/share/man/man1/java.1   /home/linuxbrew/.linuxbrew/share/man/man1/javadoc.1
/home/linuxbrew/.linuxbrew/share/man/man1/javac.1  /home/linuxbrew/.linuxbrew/share/man/man1/javap.1
```

On macOS-side, `openjdk` is keg-only so it only works after appending directory to `MANPATH` 

---

For libexec manpage directory, it should work after updating `MANPATH`, e.g.
```
export MANPATH="$MANPATH:/home/linuxbrew/.linuxbrew/opt/openjdk/libexec/man"
```

**Reply by ghost** [reply]
I haven't modified the MANPATH.
I've updated Homebrew (I'm using `linuxbrew-core`).
but when I run e.g. `man java`, no manual entry is found

Running `ls /home/linuxbrew/.linuxbrew/share/man/man1/*java*`, outputs that no such file or directory exists

**Reply by carlocab** [reply]
What's the output of `brew config` and `brew doctor`?

Does `brew update && brew reinstall openjdk` help?

**Reply by cho-m** [reply]
The issue is the `openjdk` bottle in Linuxbrew hasn't been updated yet.
It is still the bottle from June 2nd: https://github.com/Homebrew/linuxbrew-core/commit/610fb4685d8cae0f4abd02308895b91f1ca58e1b
One of maintainers of `linuxbrew-core` will need to rebuild/upload bottle with fix.

**Reply by ghost** [reply]
> What's the output of `brew config` and `brew doctor`?

They system's config is all normal and nothing is broken

> Does `brew update && brew reinstall openjdk` help?

Nope...

**Reply by carlocab** [reply]
> They system's config is all normal and nothing is broken

That's really not useful. Please show the output of `brew config`, at least.

**Reply by ghost** [reply]
Are the man pages available on your end?

**Reply by gromgit** [reply]
As @cho-m mentioned:
> The issue is the `openjdk` bottle in Linuxbrew hasn't been updated yet.

You'll have to wait for the Linuxbrew maintainers to do the necessary.

**Reply by ghost** [reply]
> You'll have to wait for the Linuxbrew maintainers to do the necessary.

Okay.
Let's keep this discussion open as the issue hasn't been resolved yet.
Also, is there a way I can help with updating of the bottle?

**Reply by carlocab** [reply]
Yep, that's what this was meant to confirm:
```
Core tap ORIGIN: https://github.com/Homebrew/linuxbrew-core
```

**Reply by ghost** [reply]
How?

**Reply by ghost** [reply]
It's okay now. 
Thanks
